components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesStorage.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.top.sites
+
+import mozilla.components.support.base.observer.Observable
+
+/**
+ * Abstraction layer above the [PinnedSiteStorage] and [PlacesHistoryStorage] storages.
+ */
+interface TopSitesStorage : Observable<TopSitesStorage.Observer> {
+    /**
+     * Adds a new pinned site.
+     *
+     * @param title The title string.
+     * @param url The URL string.
+     * @param isDefault Whether or not the pinned site added should be a default pinned site. This
+     * is used to identify pinned sites that are added by the application.
+     */
+    fun addPinnedSite(title: String, url: String, isDefault: Boolean = false)
```

Comment:
nit: `addTopSite` to match the other APIs.

```suggestion
    fun addTopSite(title: String, url: String, isDefault: Boolean = false)
```